openpaas-james/mailbox/encrypted/memory/src/main/java/com/linagora/openpaas/encrypted/InMemoryEncryptedMessageManager.scala

```diff
@@ -0,0 +1,87 @@
+import com.linagora.openpaas.pgp.Encrypter
+import javax.inject.Inject
+import javax.mail.Flags
+import org.apache.james.mailbox.MessageManager.{AppendCommand, AppendResult, MailboxMetaData}
+import org.apache.james.mailbox.model.{ComposedMessageIdWithMetaData, FetchGroup, Mailbox, MailboxACL, MailboxCounters, MailboxId, MailboxPath, MessageRange, MessageResultIterator, SearchQuery}
+import org.apache.james.mailbox.{MailboxManager, MailboxSession, MessageManager, MessageUid}
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.SFlux
+
+import scala.jdk.CollectionConverters._
+
+class InMemoryEncryptedMessageManager @Inject()(messageManager: MessageManager, keystoreManager: KeystoreManager) extends MessageManager {
+
+  override def getMessageCount(mailboxSession: MailboxSession): Long = messageManager.getMessageCount(mailboxSession)
+
+  override def getMailboxCounters(mailboxSession: MailboxSession): MailboxCounters = messageManager.getMailboxCounters(mailboxSession)
+
+  override def isWriteable(session: MailboxSession): Boolean = messageManager.isWriteable(session)
+
+  override def isModSeqPermanent(session: MailboxSession): Boolean = messageManager.isModSeqPermanent(session)
+
+  override def expunge(set: MessageRange, mailboxSession: MailboxSession): util.Iterator[MessageUid] = messageManager.expunge(set, mailboxSession)
+
+  override def search(searchQuery: SearchQuery, mailboxSession: MailboxSession): Publisher[MessageUid] = messageManager.search(searchQuery, mailboxSession)
+
+  override def delete(uids: util.List[MessageUid], mailboxSession: MailboxSession): Unit = messageManager.delete(uids, mailboxSession)
+
+  override def setFlags(flags: Flags, flagsUpdateMode: MessageManager.FlagsUpdateMode, set: MessageRange, mailboxSession: MailboxSession): util.Map[MessageUid, Flags] =
+    messageManager.setFlags(flags, flagsUpdateMode, set, mailboxSession)
+
+  override def appendMessage(msgIn: InputStream, internalDate: Date, mailboxSession: MailboxSession, isRecent: Boolean, flags: Flags): AppendResult = {
+    val keys: Seq[PublicKey] = SFlux.fromPublisher(keystoreManager.listPublicKeys(mailboxSession.getUser))
+      .collectSeq()
+      .block()
+
+    val out: ByteArrayOutputStream = new ByteArrayOutputStream
+    Encrypter.forKeys(keys.map(key => key.payload).asJava).encrypt(ByteSource.wrap(msgIn.readAllBytes()), out)
+
+    messageManager.appendMessage(new ByteArrayInputStream(out.toByteArray), internalDate, mailboxSession, isRecent, flags)
+  }
+
+  override def appendMessage(appendCommand: AppendCommand, session: MailboxSession): AppendResult = {
+    val keys: Seq[PublicKey] = SFlux.fromPublisher(keystoreManager.listPublicKeys(session.getUser))
+      .collectSeq()
+      .block()
+
+    val out: ByteArrayOutputStream = new ByteArrayOutputStream
+    Encrypter.forKeys(keys.map(key => key.payload).asJava)
+      .encrypt(ByteSource.wrap(appendCommand.getMsgIn.getInputStream.readAllBytes()), out)
```

Comment:
appendCommand.getMsgIn is a `Content` ... Content basically can be turned easily eand effectively in a byte source.

```
ByteSource contentBasedByteSource = new ByteSource() {
                @Override
                public InputStream openStream() {
                        return appendCommand.getMsgIn.getInputStream;
                }

                @Override
                public long size() {
                    return appendCommand.getMsgIn.getSize;
                }
            
```